docs/spec/spec-proposals/f1-fee-distribution/f1_fee_distr.tex

```diff
@@ -98,17 +96,21 @@ \subsection{Slashing}
 \label{ssec:slashing}
 Slashing is distinct from withdrawals, since it lowers the stake of all of the delegator's by a fixed percentage.
 Since no one is charged gas for slashes, a slash cannot iterate over all delegators.
-Thus we can no longer just multiply by $x$ over the difference in stake. 
+Thus we can no longer just multiply by $x$ over the difference in stake.
+This section describes a simple solution that should suffice for most chains needs. An asymptotically optimal solution is provided in section 2.4.
+TODO: Consider removing this section in favor of just using the current section 2.4?
+
 The solution here is to instead store each period created by a slash in the validators state.
 Then when withdrawing, you must iterate over all slashes between when you started and ended.
-Suppose you delegated at period $0$, a y\% slash occured at period $2$, and your withdrawal is period $4$.
-Then you receive funds from $0$ to $2$ as normal.
-The equations for funds you receive for $2$ to $4$ now uses $(1 - y)x$ for your stake instead of just $x$ stake.
+Suppose you delegated at period $0$, a y\% slash occured at period $2$, and your withdrawal creates period $4$.
+Then you receive funds from periods $0$ to $2$ as normal.
+The equations for funds you receive for periods $2$ to $4$ now uses $(1 - y)x$ for your stake instead of just $x$ stake.
 When there are multiple slashes, you just account for the accumulated slash factor.
 
-In practice this will not really be an efficiency hit, as we can expect most validators to have no slashes.
-Validators that get slashed a lot will naturally lose their delegators.
-A malicious validator that gets itself slashed many times would increase the gas to withdraw linearly, but the economic loss of funds due to the slashes should far out-weigh the extra overhead the honest withdrawer must pay for due to the gas.
+In practice this will not really be an efficiency hit, as the number of slashes is expected to be 0 or 1 for most validators.
+Validators that get slashed more will naturally lose their delegators.
+A malicious validator that gets itself slashed many times would increase the gas to withdraw linearly, but the economic loss of funds due to the slashes is expected to far out-weigh the extra overhead the honest withdrawer must pay for due to the gas.
+(TODO: frame that above sentence in terms of griefing factors, as thats more correct)
```

Comment:
Maybe remove all the TODOs?

Reply:
Would you prefer them to be tex comments? I would like to further polish this, and make it like a more independent algorithm description / paper.

Reply:
The algorithm hasn't / isn't changing. The parts with todos are for more clear descriptions / expositions, and are primarily on sections we don't implement.